### PR TITLE
Fixed behavior of threads type selector in configure and threads.hpp

### PR DESCRIPTION
**Pull request by joek93**
Solves threads selector issues with none and cuda threads types.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -120,8 +120,8 @@ AX_OPENMP
 
 #cuda
 AX_CHECK_CUDA
-AM_CONDITIONAL([USE_CUDA],[test "$CXX" == "nvcc"])
-if test "$CXX" == "nvcc"
+AM_CONDITIONAL([USE_CUDA],[test "$(basename $CXX)" == "nvcc"])
```
Comment (sunpho84) [inline configure.ac:123]: good

File: configure.ac [hunk 2]
```diff
@@ -120,8 +120,8 @@ AX_OPENMP
...
+if test "$(basename $CXX)" == "nvcc"
```
Comment (sunpho84) [inline configure.ac:124]: that's my bad, one should use AM_COND_IF instead

File: projects/ib/test_wip.cpp [hunk 3]
```diff
@@ -63,26 +63,26 @@ void in_main(int narg,char **arg)
...
+//    auto c=conf.getWritable();
+//    master_printf("allocated in %p\n",c._data);
+//    double& e=c[locVol-1][3][2][2][1];
+//    master_printf("end: %p, should be %p\n",&e,c._data+locVol*4*3*3*2-1);
+//  }
+//  
+//  PAR(0,locVol,
+//      CAPTURE(TO_WRITE(conf)),
+//      ivol,
+//      {
```
Comment (sunpho84) [inline projects/ib/test_wip.cpp:83]: ok

File: configure.ac (hunk 2 above)
[the remark this answers is not in the document]
Comment (joek93) [reply configure.ac:124]: Ok, makes sense

File: src/threads/threads.hpp [hunk 4]
```diff
@@ -118,7 +145,7 @@ namespace nissa
 
 #else
 
-# define CUDA_PARALLEL_LOOP(ARGS...) SERIAL_LOOP(ARGS)
+# define CUDA_PARALLEL_LOOP(ARGS...) serialFor(__LINE__,__FILE__,ARGS)
```
[the remark this answers is not in the document]
Comment (joek93) [reply src/threads/threads.hpp:148]: Maybe you already pointed out the logic in the next comment. The thing is that HOST_PARALLEL_LOOP and DEVICE_PARALLEL_LOOP appear "naked" (not built from a PAR macro) also in other parts of the code (just a few, not many). I am not sure whether that is intended, or whether one can remove those instances.

[the remark this answers is not in the document]
Comment (sunpho84) [reply src/threads/threads.hpp:148]: The whole general logic must be clarified a bit, I'll open a new PR for this.
PAR was a kind of quick hack, which uses the default parallelization, but does not take into account the actual real-case context. 

There is some logic that must be fixed before proceeding

**Comment by sunpho84** [comment]
Fixed by 46d906295c25ed4d890c84865c7da6c19235bccb

**Pull request closed without merging**